default/AI/AIstate.py

```diff
@@ -662,6 +661,18 @@ def rate_fleet(self, fleet_id, enemy_stats=None, ship_info=None):
                 shields = ship.currentMeterValue(fo.meterType.shield)
                 stats['structure'] = structure
                 stats['shields'] = shields
+                stats['attack'] = 0
+                stats['attacks'] = {}
+                design = ship.design
+                if not design:
+                    print "ERROR: Ship has no valid design :)"
+                    print "If this happened in a moderated game: Screw those. The AI simply does not work with them."
+                    continue
+                for partname in design.parts:
+                    if partname and fo.getPartType(partname).partClass == fo.shipPartClass.shortRange:
+                        damage = ship.currentPartMeterValue(fo.meterType.capacity, partname)
+                        stats['attack'] += damage
+                        stats['attacks'][damage] = stats['attacks'].get(damage, 0) + 1
```

Comment:
No, I hoped adding random entries to a dict would make all the difference. Is this not how programming works?

The fields are already defined at ```stats = dict(self.get_weighted_design_stats(design_id, species_name))``` and are used in the following lines (e.g. look at 2 lines below this comment.

I merely updated them to the correct values we want to use as design stats simply are no longer reliable due to tech upgrades and we need to query the ship meters instead.